src/ErrorHandling/ErrorModal.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from "react";
+import { makeStyles } from "tss-react/mui";
+import { Modal } from "../ui/React/Modal";
+import { Box, Button, Typography } from "@mui/material/";
+import { ErrorRecord, ErrorState } from "./ErrorState";
+import { Router } from "../ui/GameRoot";
+import { SimplePage } from "@enums";
+import { useRerender } from "../ui/React/hooks";
+import { OptionSwitch } from "../ui/React/OptionSwitch";
+import { LogBoxEvents } from "../ui/React/LogBoxManager";
+import { recentScripts } from "../Netscript/RecentScripts";
+
+export function ErrorModal(): React.ReactElement {
+  const { classes } = useStyles();
+  const rerender = useRerender();
+  const [error, setError] = useState<ErrorRecord | null>(ErrorState.ActiveError);
+
+  useEffect(() => {
+    const listener = (newError: ErrorRecord) => {
+      if (newError.force || (Router.page() !== SimplePage.ActiveScripts && !ErrorState.PreventModals)) {
+        setError(newError);
+        rerender();
+      }
+    };
+    ErrorState.ErrorUpdate.subscribe(listener);
+  }, [rerender]);
+
+  const onClose = (): void => {
+    ErrorState.ActiveError = null;
+    setError(null);
+  };
+
+  const viewLogs = (): void => {
+    onClose();
+    const recentScript = recentScripts.find((script) => script.runningScript.pid === error?.pid);
+    if (!recentScript) {
+      console.warn(`No recent script found with pid ${error?.pid}`);
+      return;
+    }
+    LogBoxEvents.emit(recentScript.runningScript);
+  };
+
+  const goToErrorPage = () => {
+    onClose();
+    ErrorState.UnreadErrors ||= 1;
+    Router.toPage(SimplePage.ActiveScripts);
+  };
+
+  return (
+    <Modal open={!!error} onClose={onClose}>
+      {error ? (
+        <Typography>
+          <h2>{error.errorType} ERROR</h2>
+          <p style={{ whiteSpace: "pre-wrap" }}>{error.message.replaceAll("/", "/​")}</p>
+          <p>Script: {error.scriptName}</p>
```

Comment:
```suggestion
          <p>
            Script: {error.scriptName}
            <br />
            PID: {error.pid}
          </p>
```
Suggestion: Show PID.